content/blog/2022/10/2022-10-04-pipeline-steps-improvement-gsoc-report.adoc

```diff
@@ -0,0 +1,104 @@
+I spent most of the community bonding period thinking about possible improvements, understanding the code base (for jenkins.io and pipeline-steps-doc-generator), and creating website wireframes that can be shared with the community. 
+This phase helped me communicate with my mentors and plan my project out, at least to a level where I could distinguish tasks for the two coding phases. 
+I created an link:https://issues.jenkins.io/browse/JENKINS-68650[epic] to generate and maintain tickets. Below is a summary of the two coding phases.
+
+== Coding Phase - I
+
+* *Updating sidebar scrolling for Jenkins documentation* (link:https://issues.jenkins.io/browse/WEBSITE-799[ticket]) -
+The sidebar required independent scrolling from the main content to improve the navigation on longer pages.
+
+* *Listing plugins on Pipeline Steps Reference* (link:https://issues.jenkins.io/browse/WEBSITE-803[ticket]) -
+I had included this idea in my proposal, but seeing the outcome, we decided that this does not add much value to the steps documentation for the users. 
+This gave rise to the idea of adding a search filter instead. See the discussion in this link:https://github.com/jenkins-infra/jenkins.io/pull/5245[PR].
+
+* *Adding search filter* (link:https://issues.jenkins.io/browse/WEBSITE-807[ticket]) -
+There are more than 1500 steps listed on the Pipeline Steps Reference page, and filtering them becomes a muchrequired feature for the users. 
+With this change, users can now type in any string, and the content of the page is changed dynamically according to the input.
+
+* *Separating declarative steps generation from main class* (link:https://issues.jenkins.io/browse/JENKINS-68811[ticket]) -
+A small but helpful change was to separate all the functions that generate declarative Pipeline steps from the main (`PipelineStepExtractor`) class. 
+This would not only make the main class more readable but also isolate the specific functions related to declarative steps, making the code more modular.
+
+* *Shifting parameter data types* (link:https://issues.jenkins.io/browse/WEBSITE-801[ticket]) -
+This was an initiative to reduce the overall length of each plugin page listed on Pipeline Steps Reference. 
+The idea was to move the parameter data type (an essential piece of information regarding a step's parameter that a user might want to know) from the bottom of the help text to inline with the name of the parameter/property.
+
+* *Releasing pipeline-metadata-utils* (link:https://issues.jenkins.io/browse/WEBSITE-806[ticket]) -
+`pipeline-metadata-utils` is a tool that can be imported onto a project as a maven dependency. It provides a `HyperLocalPluginManager` that can be used to query plugins and build documentation with its help. My work involved updating and refining the code base and writing tests for the same. The released artifact is linked link:https://repo.jenkins-ci.org/ui/repos/tree/General/releases/org/jenkins-ci/infra/pipeline-metadata-utils[here].
```

Comment:
Usually best to avoid the word "here" as a hyperlink.  Better to describe the destination of the link.
```suggestion
`pipeline-metadata-utils` is a tool that can be imported onto a project as a maven dependency. It provides a `HyperLocalPluginManager` that can be used to query plugins and build documentation with its help. My work involved updating and refining the code base and writing tests for the same. The released artifact is available in the Jenkins link:https://repo.jenkins-ci.org/ui/repos/tree/General/releases/org/jenkins-ci/infra/pipeline-metadata-utils[artifact repository].
```